src/storage-client/src/controller.rs

```diff
@@ -1084,13 +1082,28 @@ where
         let mut to_create = Vec::with_capacity(to_register.len());
         // This work mutates the controller state, so must be done serially. Because there
         // is no io-bound work, its very fast.
-        for (id, description, write, since_handle, collection_state) in to_register {
-            self.state.persist_write_handles.register(id, write);
-            self.state.persist_read_handles.register(id, since_handle);
 
-            self.state.collections.insert(id, collection_state);
+        {
+            // We hold this lock for a very short amount of time, just doing some hashmap inserts
+            // and unbounded channel sends.
+            let mut source_statistics = self.state.source_statistics.lock().expect("poisoned");
+            for (id, description, write, since_handle, collection_state) in to_register {
+                self.state.persist_write_handles.register(id, write);
+                self.state.persist_read_handles.register(id, since_handle);
+
+                self.state.collections.insert(id, collection_state);
+
+                if let DataSource::Ingestion(i) = &description.data_source {
+                    source_statistics.insert(id, statistics::StatsInitState::Uninitialized);
```

Comment:
Instead of adding `StatsInitState`, couldn't we just keep the previous type of the field, and `insert()` an empty `HashMap` for `id` here. Then, when we receive statistics updates, we only put them in the `HashMap` when there is one, for the given id, otherwise we ignore it.

And we drop the `HashMap` for a given `id` when we process the "drop" messages.

@petrosagg and you are more rustaceous than I am, so it's hard for me to make the call, but it seems that `StatsInitState` is more than what we need.

Reply:
We can definitely do this, but it was becoming exceedingly difficult to reason about; Originally I actually thought I needed 3 different states, which is why I added the enum, but I decided to keep it because it let me encapsulate some logic

Reply:
Yeah i thought about it, and I think `Uninitialized` and the logic surrounding it is not required. Basically, if we get an update after the data is removed, we can 

Updated the pr to reflect this!